File: FizzBuzz/src/test/java/kata/game/fizzbuzz/FizzBuzzTest.java
package kata.game.fizzbuzz;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

public class FizzBuzzTest {

    private static final String TEST_NOT_IMPLEMENTED_MESSAGE = "<test not implemented>";
    private FizzBuzz testFizzBuzz;

    @Before
    public void before() throws Exception {
        testFizzBuzz = new FizzBuzz();
    }

    @After
    public void after() throws Exception {
    }

    @Test
    public void whenNumberDivadableByThree_ThenFizz() throws Exception {
        String result = testFizzBuzz.getResult(6);
        assertThat(result, equalTo(FizzBuzzConstants.FIZZ));
    }

    @Test
    public void whenNumberDivadableByFive_ThenBuzz() throws Exception {
        String result = testFizzBuzz.getResult(10);
        assertThat(result, equalTo(FizzBuzzConstants.BUZZ));
    }

    @Test
    public void whenNumberDivadableByThreeAndFive_ThenFizzBuzz() throws Exception {
        String result = testFizzBuzz.getResult(15);
        assertThat(result, equalTo(FizzBuzzConstants.FIZZBUZZ));
    }

    @Test
    public void whenNumberNotDivadableByThreeOrFive_ThenGetNumberAsString() throws Exception {
        int testNumber = 11;
        String result = testFizzBuzz.getResult(testNumber);
        assertThat(result, equalTo(Integer.toString(testNumber)));
    }
} 

File: FizzBuzz/Golang/src/fizzbuzz_lib/fizzbuzz_test.go
package fizzbuzz_lib

import (
	"testing"
	"strconv"
)

 func TestFizz(t *testing.T) {
	 var number int = 3
	 if GetResult(number) != "Fizz" {
		 t.Errorf("%d should be Fizz", number)
	 }
 }

func TestBuzz(t *testing.T) {
	var number int = 5
	if GetResult(number) != "Buzz" {
		t.Errorf("%d should be Buzz", number)
	}
}

func TestFizzBuzz(t *testing.T) {
	var number int = 3 * 5
	if GetResult(number) != "FizzBuzz" {
		t.Errorf("%d should be FizzBuzz", number)
	}
}

func TestNumber(t *testing.T) {
	var number int = 2
	if GetResult(number) != strconv.Itoa(number) {
		t.Errorf("%d should be \"%d\"", number, number)
	}
}
